src/main/java/com/Elessar/app/server/MyServer.java

```diff
@@ -23,22 +28,23 @@
 
     private final String serverName;
     private final int port;
-    private final Map<String, User> userData;
+    private final MyDatabase users;
 
     public MyServer(String serverName, int port) {
         this.serverName = serverName;
         this.port = port;
-        userData = new HashMap<String, User>();
+        MongoClient mongoClient = MongoClients.create("mongodb://localhost:27017");
+        users = new MongoDB(mongoClient.getDatabase("myDB").getCollection("users"));
```

Comment:
You should pass a MyDatabase to the constructor instead of creating the instance inside, e.g.
```java
public MyServer(String serverName, int port, MyDatabase db) {
  ...
}
```

Doing this allows us to switch to a different implementation of the database without any code change for MyServer class.